kits/arm/arm.py

```diff
@@ -0,0 +1,217 @@
+            self.cmd.read_gains(gains_file)
+            if not self.grp.send_command_with_acknowledgement(self.cmd):
+                raise RuntimeError('Did not receive ack from group.')
+            print('Successfully read gains from file and sent to module.')
+        except Exception as e:
+            print('Problem reading gains from file or sending to module: {0}'.format(e))
+    
+    
+    # Updates feedback and generates the basic command for this timestep
+    def update(self):
+        self.prev_fbk = list(self.fbk)
+        self.grp.get_next_feedback(reuse_fbk=self.fbk)
+        self.time_now = perf_counter()
+        if not self.at_goal:
+            self.t_traj = self.time_now - self.traj_start_time
+            if self.time_now > (self.traj_start_time + self.duration):
+                self.at_goal = True
+        # Create command
+        if self.goal == "grav comp":
+            # Calculate required torques to negate gravity at current position
+            nan_array = np.empty(self.grp.size)
+            nan_array[:] = np.nan
+            self.cmd.position = nan_array
+            self.cmd.velocity = nan_array
+            self.cmd.effort = get_grav_comp_efforts(self.model, self.fbk.position, -self.gravity_vec)
+        elif not self.at_goal:
+            # Move to target position
+            self.pos_cmd, self.vel_cmd, self.accel_cmd = self.trajectory.get_state(self.t_traj)
+            self.cmd.position = self.pos_cmd
+            self.cmd.velocity = self.vel_cmd
+            self.cmd.effort = get_grav_comp_efforts(self.model, self.fbk.position, -self.gravity_vec)
+        else:
+            # Stay at current position
+            self.cmd.position = self.fbk.position_command
+        return 
+    
+    
+    def send(self):
+        # Send command
+        self.grp.send_command(self.cmd)
+        return
+
+    
+    def createMotionArray(self, size, prev_cmd, array=None, flow=None):
+        # Helper function to create arrays for things like waypoints, velocities, and accelerations that are used in creating a trajectory
+        # Note flow overrides any array passed (used for vel and accel, should never pass flow and positions)
+        mArray = np.empty((self.grp.size, size+1))
+        mArray[:, 0] = prev_cmd
+        if flow == None:
+            for i in range(size):
+                if array == None:
+                    mArray[:, i+1] = np.zeros(self.grp.size)
+                else:
+                    mArray[:, i+1] = array[i]
+        else:
+            nan_array = np.empty(self.grp.size)
+            nan_array[:] = np.nan
+            for i in range(size):
+                if flow[i]:
+                    mArray[:, i+1] = nan_array
+                else:
+                    mArray[:, i+1] = np.zeros(self.grp.size)
+        
+        
+        return mArray
+    
+    
+    def createGoal(self, position, durration=None, velocity=None, accel=None, flow=None):
+        # Create trajectory to target position
```

Comment:
Thanks for writing up documentation on the parameters here. One thing though:

We should do this whole comment as a multi line comment (like the `Ex: ...` below it), meaning this should be replaced with:

```py
  def create_goal(params....):
    """
    Create trajectory to target position
    ... Blah Blah ...
    ......
    Ex:
        3dof arm with a 2 point trajectory
         .....
```